app/assets/javascripts/uploadcare/widget/tabs/preview-tab.coffee

```diff
@@ -39,27 +40,35 @@ uploadcare.namespace 'widget.tabs', (ns) ->
         source = info.sourceInfo
         blob = utils.abilities.Blob
         if source.file and blob and source.file instanceof blob
-          tryToLoadImage(file, source.file)
+          tryToLoadImagePreview(file, source.file)
+          .fail =>
+            tryToLoadVideoPreview(file, source.file)
 
       @file.done ifCur (info) =>
         state = if info.isImage then 'image' else 'regular'
-        if state != 'image' or state != @__state
+        if info.sourceInfo.source == 'camera'
+          state = 'video'
```

Comment:
This is strange. if `info.sourceInfo.source` == 'camera', it can be both image or video.